backend/webserver/views.py

```diff
@@ -239,6 +242,50 @@ def get(self, request):
         return Response(serializer.data, status=status.HTTP_200_OK)
 
 
+class LikedPostView(APIView):
+    authentication_classes = [BasicAuthentication]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    
+    def get(self,requst,pk,*args,**kwargs):
+        author = self.get_author(pk)
+        public_post = Post.objects.filter(visibility="PUBLIC").all()
+        liked = Like.objects.filter(liker=author.id).filter(post__in=public_post.values_list('id')).all()
+        serializer = LikePostSerializer(liked, many=True, context={'request': requst})
+        # I am doing this so that I can display the post like this "object":"http://127.0.0.1:5454/authors/9de17f29c12e8f97bcbbd34cc908f1baba40658e/posts/764efa883dda1e11db47671c4a3bbd9e"
+        for data in serializer.data:
+            data['post'] = data['liker']['url'] + "posts/"+ str(data['post'])+ "/" 
+        return Response(serializer.data, status=status.HTTP_200_OK)
+        
+
+
+class LikePostView(APIView):
+    authentication_classes = [BasicAuthentication]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    def get_post(self,post_id,author_id):
+        post = get_object_or_404(Post,id=post_id,author=author_id)
+        return post
+    
+    def get(self,requst,pk,post_id,*args,**kwargs):
+        author = self.get_author(pk)
+        post = self.get_post(author_id=author.id,post_id=post_id)
+        likes = Like.objects.filter(post=post.id).all()
+        serializer = LikePostSerializer(likes, many=True, context={'request': requst})
+        
+        # I am doing this so that I can display the post like this "object":"http://127.0.0.1:5454/authors/9de17f29c12e8f97bcbbd34cc908f1baba40658e/posts/764efa883dda1e11db47671c4a3bbd9e"
+        for data in serializer.data:
+            data['post'] = data['liker']['url'] + "posts/"+ str(data['post'])+ "/"  
```

Comment:
```suggestion
            data['post'] = urljoin(data['liker']['url'].strip('/'),  f"posts/{data['post']}/")
```
Manually joining urls can be error-prone and unsafe. Use this function from urllib.parse - https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin